spt/cvars.cpp

```diff
@@ -325,59 +198,140 @@ void Cvar_RegisterSPTCvars()
-			RemoveCommandFromList(pluginCommandListHead, todelete);
+			RemoveCommandFromList(todelete);
 			continue;
 		}
+		else
+		{
+#ifdef OE
+			if (cmd->IsFlagSet(FCVAR_HIDDEN))
+				spt_cvarstuff.hidden_cvars.push_back(cmd);
+#endif
+			cmd = (ConCommandBase*)cmd->GetNext();
+		}
+
 		HandleBackwardsCompatibility(*inittedCmd, cmdName);
+	}
+
+	for (auto& featCmd : cmd_to_feature)
+	{
+		auto regcmd = featCmd.command;
+		if (strlen(regcmd->GetName()) > 0) {
 #ifdef OE
-		if (cmd->IsBitSet(FCVAR_HIDDEN))
-			spt_cvarstuff.hidden_cvars.push_back(cmd);
+			regcmd->AddFlags(FCVAR_PLUGIN);
+			// Unlink from plugin list
+			// Necessary because RegisterConCommandBase skips the command if it's next isn't null
+			regcmd->SetNext(NULL);
 #endif
-		cmd = (ConCommandBase*)cmd->GetNext();
+			g_pCVar->RegisterConCommand(regcmd);
+			if (regcmd->m_bRegistered) {
+				Msg("command %s was registered\n", regcmd->GetName());
+			}
+		}
+	}
+}
+
+static ConCommandBase** GetGlobalCommandListHead()
+{
+	// According to the SDK, ICvar::GetCommands is position 9 on the vtable
+	void** icvarVtable = *(void***)g_pCVar;
+	uint8_t* addr = (uint8_t*)icvarVtable[9];
+	// Follow along thunked function
+	if (*addr == X86_JMPIW)
+	{
+		uint32_t offset = *(uint32_t*)(addr + 1);
+		addr += x86_len(addr) + offset;
+	}
+	// Check if it's the right instruction
+	if (*addr == X86_MOVEAXII)
+		return *(ConCommandBase***)(addr + 1);
+	return NULL;
+}
+
+static void RemoveCommandFromList(ConCommandBase** head, ConCommandBase* command)
+{
+	if (head == NULL)
+		return;
+
+	ConCommandBase* pPrev = NULL;
+	for (ConCommandBase* pCommand = *head; pCommand;
+		pCommand = pCommand->m_pNext)
+	{
+		if (pCommand != command)
+		{
+			pPrev = pCommand;
+			continue;
+		}
+
+		if (pPrev == NULL)
+			*head = pCommand->m_pNext;
+		else
+			pPrev->m_pNext = pCommand->m_pNext;
+
+		pCommand->m_pNext = NULL;
+		break;
 	}
-#ifndef OE
-	ConVar_Register(0);
+}
+
+static void UnregisterConCommand(ConCommandBase* pCommandToRemove)
+{
+#ifdef OE
+	static bool searchedForList = false;
+	static ConCommandBase** globalCommandListHead = nullptr;
+	Msg("Unregistering %s\n", pCommandToRemove->GetName());
+
+	// Look for the global command list head once
+	if (!searchedForList)
+	{
+		globalCommandListHead = GetGlobalCommandListHead();
+		searchedForList = true;
+	}
+
+	if (!globalCommandListHead)
+		return;
+
+	pCommandToRemove->m_bRegistered = false;
+	RemoveCommandFromList(globalCommandListHead, pCommandToRemove);
 #else
-	ConCommandBaseMgr::OneTimeInit(&g_ConVarAccessor);
-	// Clear the list head like newer engines do and reset s_pAccessor so the list doesn't break
-	// next time SPT commands are registered
-	*pluginCommandListHead = NULL;
-	ConCommandBaseMgr::OneTimeInit(NULL);
+	g_pCVar->UnregisterConCommand(pCommandToRemove);
 #endif
-
-	first_time_init = false;
 }
 
+
 void Cvar_UnregisterSPTCvars()
 {
-	if (!interfaces::g_pCVar)
+	if (!g_pCVar)
 		return;
 
+	ConCommandBase::s_pConCommandBases = NULL;
```

Comment:
If you care about mimicking what the engine does naturally, this should be moved to the end of of Cvar_RegisterSPTCvars. Atm we are kinda living with a somewhat bogus pointer in this variable through the lifetime of the plugin since it's never cleared

Reply:
The reasoning for this is that in OE the plugin list contains (at least?) one concommand after the cvars have been loaded normally. This caused a crash upon reloading the plugin